lib/writer.js

```diff
@@ -38,6 +38,7 @@ function Writer(opts) {
   this.opts = utils.address(opts);
   this.conns = new Set;
   this.pendingConns = new Set;
+  this.publishQueue = [];
```

Comment:
since nsq.js is already using `Set`, and there are comments elsewhere about possible deduplication, does it make sense to use `Set` for this as well?

there was another comment about deduplication, `Set` does that for us.

Reply:
Order matters and duplicates shouldn't be removed.

Reply:
It wouldn't work with the way we are queueing the messages (arrays would only dedupe if they had the same reference), and I'm also not sure if we want to dedupe anything. The queue is a temporary state while nsq tries to reconnect. The publish command will return the success or failure of the publish request, and it will not return anything while the request is queued.